### DISCUSSION TITLE
Image manipulation doesn't work with saveToS3()

### DISCUSSION THREAD

**Discussion by simaodiascosta** [stefanzweifel/sidecar-browsershot#70] [Ideas]
I'm trying to use `->fit()` that is [documented in spatie](https://spatie.be/docs/browsershot/v2/usage/creating-images?query=savetos3#content-sizing-the-image) but the output is not showing what is expected.

My code is:

```
$windowWidth = 2160;
$windowHeight = 3840;
$fitWidth = 1080;
$fitHeight = 1920;

BrowsershotLambda::url($url)
    ->waitUntilNetworkIdle()
    ->windowSize($windowWidth, $windowHeight)
    ->fit(Manipulations::FIT_FILL_MAX, $fitWidth, $fitHeight)
    ->setScreenshotType('jpeg', 80)
    ->saveToS3($filePath, $s3Disk);
```

Looking at `save()` in Browsershot class, [I can see](https://github.com/spatie/browsershot/blob/8454698101b78c78885bb712b37cd1da97ac8e2e/src/Browsershot.php#L586) that `applyManipulations()` is called while in BrowsershotLambda `saveToS3()` such thing [doesn't seem to exist](https://github.com/stefanzweifel/sidecar-browsershot/blob/360fbeac1ba7ff2715145f2a5da866e6bdcc998d/src/BrowsershotLambda.php#L63).

Thanks for looking at this issue 👍

**Comment by stefanzweifel** [comment]
Thanks for bringing this up.
Maybe calling `applyManipulations()` in `saveToS3()` is enough, but I guess we then would need to run Imagick on Lambda? 🤔 

Need to take a closer look at this over the coming days/weeks.

**Reply by simaodiascosta** [reply]
Thanks for looking into this 🙌

**Reply by stefanzweifel** [reply]
@simaodiascosta I've prepared a pull request that fixes this issue in #71.

To apply the manipulations, sidecar-browsershot will download the image from S3 to a local directory, apply the manipulations and then re-upload the image back to S3.

Would that fit into your workflow? I don''t see any other way right now to solve this. The image has to be downloaded somewhere, where PHP is executed.

**Reply by simaodiascosta** [reply]
Yes it does fit my workflow.

Really appreciate you looking at this so quickly 🙏

**Reply by stefanzweifel** [reply]
I've updated the PR to highlight in the README, that the package downloads the image in the background and does the manipulation in the background.

PR has been merged and a new version has been released: https://github.com/stefanzweifel/sidecar-browsershot/releases/tag/v1.9.0
